src/REVLoans.sol

```diff
@@ -492,16 +504,18 @@ contract REVLoans is ERC721, ERC2771Context, IREVLoans, ReentrancyGuard {
                     beneficiary: payable(owner),
                     controller: controller
                 });
+
+                isCollateralReturned = true;
             }
 
             // Decrement the amount loaned.
             totalBorrowedFrom[revnetId][loan.source.terminal][loan.source.token] -= loan.amount;
 
             // Decrement the total amount of collateral tokens supporting loans from this revnet.
-            totalCollateralOf[revnetId] -= loan.collateral;
+            if (!isCollateralReturned) totalCollateralOf[revnetId] -= loan.collateral;
```

Comment:
can we just do this -= in line 508 where isCollateralReturned is set to true? this was we can remove `isCollateralReturned` all together maybe?

Reply:
no, if we did, normal liquidations wouldn't result in accurate totalCollateralOf.